### PR TITLE
feat: support customEmit ncc option

### PR DESCRIPTION
**Pull request by guybedford**
This adds the `customEmit` option from https://github.com/vercel/webpack-asset-relocator-loader/pull/111 to ncc directly allow custom asset emission hooks.

This PR is pending that update so the tests will be failing until then.

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -77,7 +81,7 @@ for (const unitTest of fs.readdirSync(`${__dirname}/unit`)) {
   });
 }
 for (const cliTest of eval(fs.readFileSync(__dirname + "/cli.js").toString())) {
-  it(`should execute "ncc ${(cliTest.args || []).join(" ")}"`, async () => {
+  it.skip(`should execute "ncc ${(cliTest.args || []).join(" ")}"`, async () => {
```
Comment (styfle) [inline test/index.test.js:84]: Why skip?

File: test/index.test.js [hunk 2]
```diff
@@ -134,7 +138,7 @@ for (const integrationTest of fs.readdirSync(__dirname + "/integration")) {
   // disabled pending https://github.com/zeit/ncc/issues/141
   if (integrationTest.endsWith('loopback.js')) continue;
 
-  it(`should execute "ncc run ${integrationTest}"`, async () => {
+  it.skip(`should execute "ncc run ${integrationTest}"`, async () => {
```
Comment (styfle) [inline test/index.test.js:141]: Let's revert the skip

File: test/index.test.js [hunk 3]
```diff
@@ -165,7 +169,7 @@ for (const integrationTest of fs.readdirSync(__dirname + "/integration")) {
   });
 }
 
-it(`should execute "ncc build web-vitals" with target config`, async () => {
+it.skip(`should execute "ncc build web-vitals" with target config`, async () => {
```
Comment (styfle) [inline test/index.test.js:172]: Same here

File: test/index.test.js (hunk 1 above)
Comment (guybedford) [reply test/index.test.js]: Nicely spotted, thanks.

**Review by styfle — APPROVED** [review]

**Pull request merged**